mri/reconstruct/fourier.py

```diff
@@ -208,3 +208,139 @@ def adj_op(self, x):
         """
         self.plan.f = x
         return np.copy(self.plan.adjoint()) / np.sqrt(self.plan.M)
+
+
+class Stacked3D:
+    """"  3-D non uniform Fast Fourrier Transform class,
+        fast implementation for Stacked samples
+    Attributes
+    ----------
+    samples: np.ndarray
+        the mask samples in the Fourier domain.
+    shape: tuple of int
+        shape of the image (necessarly a square/cubic matrix).
+    fft_type: 'NFFT' | 'NUFFT' default 'NFFT'
+        What FFT version to be used for 2D non uniform FFT
+    platform: string, 'cpu', 'multi-cpu' or 'gpu' default 'gpu'
+        string indicating which hardware platform will be used to compute the
+        NUFFT. works only if fft_type=='NUFFT'
+    """
+
+    def __init__(self, samples, shape, fft_type='NFFT', platform='gpu'):
+        """ Init function for Stacked3D class
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the Fourier domain.
+        shape: tuple of int
+            shape of the image (necessarly a square/cubic matrix).
+        fft_type: 'NFFT' | 'NUFFT' default 'NFFT'
+            What FFT version to be used for 2D non uniform FFT
+        platform: string, 'cpu', 'multi-cpu' or 'gpu' default 'gpu'
+            string indicating which hardware platform will be used to
+            compute the NUFFT. works only if fft_type=='NUFFT'
+        """
+        self.num_slices = shape[2]
+        self.shape = shape
+        # Sort the incoming data based on Z, Y then X coordinates
+        # This is done for easier stacking
+        self.sort_pos = np.lexsort(tuple(samples[:, i]
+                                         for i in np.arange(3)))
+        samples = samples[self.sort_pos]
+        plane_samples, self.z_samples = self.get_stacks(samples)
+        if fft_type == 'NUFFT':
+            self.FT = NUFFT(samples=plane_samples, shape=shape[0:2],
+                            platform=platform)
+        elif fft_type == 'NFFT':
+            self.FT = NFFT(samples=plane_samples, shape=shape[0:2])
+
+    def get_stacks(self, samples):
+        """ Function that converts an incoming 3D kspace samples
+            and converts to stacks of 2D. This function also checks for
+            any issues incoming k-space pattern and if the stack property
+            is not satisfied.
+            Stack Property:
+                The k-space locations originate from a stack of 2D samples
+        Parameters
+        ----------
+        samples: np.ndarray
```

Comment:
I think they are not samples or mask samples right? they are sample locations.
I think this is in-line with what @LElgueddari  said in this [comment](https://github.com/CEA-COSMIC/pysap-mri/pull/34/files#r326042782), but it happens throughout the code (both in docs and in variable names)

Reply:
This seems to be like a legacy which I followed. The idea is that from all the Fourier operators viewpoint, samples mean where the samples come from. 
AT a point this will become a philosophical conversation :P , but we can decide what we want to name it and I could fix it at all places in this same PR with string replace.
@LElgueddari @zaccharieramzi , to comment

Reply:
can you point me to where you saw that "samples mean where the samples come from".
I think it's not a philosophical conversation. To me the difference between samples and sample locations is pretty clear, and confusing both is often dangerous or misleading.

Reply:
Will open a newe issue for this, I will not cover this in this PR

Reply:
why?

Reply:
The PR is very bulky already. I feel it makes sense to handle this as a separate issue

Reply:
I don't understand. To me you are introducing a misleading variable name, you need to correct it. It doesn't change the bulk of this PR, just corrects it.

Reply:
Well, I can fix in this Class surely, but across to all other classes too? That is what I am saying will not be done in this PR

Reply:
yes not necessarily in other classes just start here. What are these other classes?

Reply:
NFFT / NUFFT all use samples as a parameter